addons/sourcemod/scripting/nd_discord_playerlist.sp

```diff
@@ -0,0 +1,193 @@
+#include <sdktools>
+#include <nd_stocks>
+#include <nd_team_eng>
+
+Handle SaveTeamsTimer;
+bool SavingTeams;
+Handle DatabaseHandle;
+char PlayerNames[MAXPLAYERS + 1][MAX_NAME_LENGTH];
+
+public Plugin myinfo =
+{
+    name        = "[ND] Discord Player List",
+    author      = "zookatron",
+    description = "Provides player list information to Discord",
+    version     = "dummy",
+    url         = "https://github.com/stickz/Redstone/"
+};
+
+public void OnPluginStart()
+{
+    if (!SQL_CheckConfig("discord"))
+    {
+        SetFailState("Could not find configuration for \"discord\" database");
+        return;
+    }
+
+    SQL_TConnect(OnDatabaseConnected, "discord");
+}
+
+public void ND_OnPlayerTeamChanged(int client, bool valid)
+{
+    SaveTeams();
+}
+
+public void OnClientPutInServer(int client)
+{
+    GetClientName(client, PlayerNames[client], MAX_NAME_LENGTH);
+    SaveTeams();
+}
+
+public void OnClientDisconnect_Post(int client)
+{
+    PlayerNames[client][0] = '\0';
+    SaveTeams();
+}
+
+public OnDatabaseConnected(Handle owner, Handle handle, const char[] error, any:data)
+{
+    if (!IsValidHandle(handle))
+    {
+        SetFailState("Unable to connect to \"discord\" database: %s", error);
+        return;
+    }
+
+    DatabaseHandle = handle;
+
+    char query[1024];
+    FormatEx(query, sizeof(query), "SELECT * FROM players;");
+    SQL_TQuery(DatabaseHandle, Setup_Step1_CheckTable, query);
+}
+
+void Setup_Step1_CheckTable(Handle owner, Handle handle, const char[] error, any:data)
+{
+    // Create the table if it doesn't exist
+    if (error[0] != '\0')
```

Comment:
Rather than trigger the error the first time through there is an alternative method where the SQL query can be modified to run "CREATE TABLE IF NOT EXISTS "

Reply:
done